Tests/Pcap++Test/Tests/LiveDeviceTests.cpp

```diff
@@ -379,91 +380,209 @@ PTF_TEST_CASE(TestPcapLiveDeviceStatsMode)
+
+		// sanity - test blocking mode returns before timeout
+		int packetCount = 0;
+		PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeNoTimeout, &packetCount, 30), 1);
+		PTF_ASSERT_EQUAL(packetCount, 5);
+
+		// verify stop capture doesn't do any effect on blocking mode
+		liveDev->stopCapture();
+		PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeTimeout, nullptr, 1), -1);
+		packetCount = 0;
+		PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeNoTimeout, &packetCount, 30), 1);
+		PTF_ASSERT_EQUAL(packetCount, 5);
+
+		// verify it's possible to capture non-blocking mode after blocking mode
+		packetCount = 0;
+		PTF_ASSERT_TRUE(liveDev->startCapture(packetArrives, &packetCount));
+
+		int totalSleepTime = 0;
+		while (totalSleepTime <= 5)
+		{
+			pcpp::multiPlatformSleep(1);
+			totalSleepTime += 1;
+			if (packetCount > 0)
+				break;
+		}
+
+		liveDev->stopCapture();
+
+		PTF_PRINT_VERBOSE("Total sleep time: " << totalSleepTime << " secs");
+
+		PTF_ASSERT_GREATER_THAN(packetCount, 0);
+
+		// verify it's possible to capture blocking mode after non-blocking mode
+		packetCount = 0;
+		PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeNoTimeout, &packetCount, 30), 1);
+		PTF_ASSERT_EQUAL(packetCount, 5);
+
+		// try to start capture from within the callback, verify no error
+		packetCount = 0;
+		PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeStartCapture, &packetCount, 30), 1);
+		PTF_ASSERT_EQUAL(packetCount, 5);
+
+		// try to stop capture from within the callback, verify no impact on capturing
+		packetCount = 0;
+		PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeStopCapture, &packetCount, 10), 1);
+		PTF_ASSERT_EQUAL(packetCount, 5);
+
+		// verify it's possible to capture non-blocking after the mess done in previous lines
+		packetCount = 0;
+		PTF_ASSERT_TRUE(liveDev->startCapture(packetArrives, &packetCount));
+
+		// verify an error returns if trying capture blocking while non-blocking is running
+		pcpp::Logger::getInstance().suppressLogs();
+		PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeTimeout, nullptr, 1), 0);
+		pcpp::Logger::getInstance().enableLogs();
+
+		totalSleepTime = 0;
+		while (totalSleepTime <= 5)
+		{
+			pcpp::multiPlatformSleep(1);
+			totalSleepTime += 1;
+			if (packetCount > 0)
+				break;
+		}
+
+		PTF_PRINT_VERBOSE("Total sleep time: " << totalSleepTime << " secs");
+
+		liveDev->stopCapture();
+		PTF_ASSERT_GREATER_THAN(packetCount, 0);
+
+		liveDev->close();
+
+		// a negative test
+		pcpp::Logger::getInstance().suppressLogs();
+		PTF_ASSERT_FALSE(liveDev->startCapture(packetArrives, &packetCount));
+		pcpp::Logger::getInstance().enableLogs();
+	}
+} // TestPcapLiveDeviceBlockingMode
+
+PTF_TEST_CASE(TestPcapLiveDeviceBlockingModePollTimeout)
+{
+#if !defined(_WIN32) and defined(MANUAL_TEST)
```

Comment:
although this test cannot run on CI, we can manually run it locally.
I think it is worth keeping it, so I added a `MANUAL_TEST` flag for it.

Reply:
I found that it's really difficult to use iptables to block all packets. No matter how I set up rules, I still can see packets on Wireshark. Maybe the easiest way is to use a clean interface to test... since we can ensure no packets on it ...

Reply:
@seladb  please let me know your opinion

Reply:
Yes, I also encountered problems with `iptables`. I think we should just remove these tests. There is no point in `MANUAL_TEST` which we don't run anywhere...